### ISSUE TITLE
jax.scipy.special.erfcx returns spurious 0.0 in a band near x≈26.6 (should be ~0.0212; scipy/torch agree)

### ISSUE THREAD

**Issue by OE-GOD** [jax-ml/jax#38607]
### Description

`jax.scipy.special.erfcx(x)` returns exactly `0.0` across a narrow band of large `x` (approximately `x ∈ [26.54, 26.64]`), where the true value is ≈ `0.0212`:

```python
import jax; jax.config.update('jax_enable_x64', True)
import jax.scipy.special as jsp, scipy.special as ssp
print(jsp.erfcx(26.6), ssp.erfcx(26.6))   # 0.0   vs 0.02119517815916648
# neighbours are correct:
print(jsp.erfcx(26.0), jsp.erfcx(27.0))    # 0.021683..., 0.020881...  (match scipy)
```

`erfcx` is strictly positive for all finite `x` (asymptotically `1/(x*sqrt(pi))` ≈ `0.02121` at `x=26.6`), so `0.0` is wrong. The band reproduces under eager, `jit`, and `vmap`. Neighboring points (26.0, 27.0, 30.0) are all correct, so this is a localized incorrect-zero band rather than a broad asymptotic error — consistent with `erfc(x)` underflowing to `0` inside a composed `exp(x**2) * erfc(x)` implementation instead of using the scaled asymptotic form.

### Cross-check

`scipy.special.erfcx(26.6)` and `torch.special.erfcx(26.6)` both return `0.02119517815916648`; the large-`x` asymptote `1/(x*sqrt(pi))` confirms it. JAX is the lone outlier.

(The gradient is affected downstream: `jax.grad(erfcx)(26.6)` returns `-2/sqrt(pi)` because the vanishing forward value zeroes the `2*x*erfcx(x)` term; this resolves once the value is fixed.)

### System info
```
jax 0.10.1, jaxlib 0.10.1, numpy 2.4.6, python 3.12.13, CPU
```

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
